alibi_detect/cd/tensorflow/classifier.py

```diff
@@ -120,22 +115,38 @@ def score(self, x: np.ndarray) -> float:
 
         Returns
         -------
-        Drift metric (e.g. accuracy) obtained from out-of-fold predictions from a trained classifier.
+        p-value, and a notion of distance between the trained classifier's out-of-fold performance
+        and that which we'd expect under the null assumption of no drift.
         """
         x_ref, x = self.preprocess(x)
         x, y, splits = self.get_splits(x_ref, x)
 
         # iterate over folds: train a new model for each fold and make out-of-fold (oof) predictions
-        preds_oof, idx_oof = [], []
+        preds_oof_list, idx_oof_list = [], []
         for idx_tr, idx_te in splits:
             x_tr, y_tr, x_te = x[idx_tr], np.eye(2)[y[idx_tr]], x[idx_te]
             clf = tf.keras.models.clone_model(self.model)
             clf.compile(**self.compile_kwargs)
             clf.fit(x=x_tr, y=y_tr, **self.train_kwargs)
             preds = clf.predict(x_te, batch_size=self.train_kwargs['batch_size'])
-            preds_oof.append(preds)
-            idx_oof.append(idx_te)
-        preds_oof = np.concatenate(preds_oof, axis=0)[:, 1]
-        idx_oof = np.concatenate(idx_oof, axis=0)
-        drift_metric = self.metric_fn(y[idx_oof], preds_oof)
-        return drift_metric
+            preds_oof_list.append(preds)
+            idx_oof_list.append(idx_te)
+        preds_oof = np.concatenate(preds_oof_list, axis=0)[:, 1]
+        idx_oof = np.concatenate(idx_oof_list, axis=0)
+
+        if self.soft_preds:
+            log_losses_ref = preds_oof[y[idx_oof] == 0]
```

Comment:
Maybe need to be careful here as, contrary to PyTorch, TensorFlow models often return softmax output instead of logits.

Reply:
The KS-test stat will remain the same for any monotonic transformation of the data (so doesn't matter if probs, log-probs, logits etc) but maybe I should use more general name to avoid confusion.

Reply:
Actually I'm not sure if it's a great idea to apply it directly to logits. I'm going to change to ensure it's applied to probs